src/schrodinger/sketcher/dialog/paste_in_text_dialog.cpp

```diff
@@ -1,12 +1,78 @@
 #include "paste_in_text_dialog.h"
 
+#include <algorithm>
+#include <tuple>
+#include <unordered_map>
+#include <vector>
+
+#include "schrodinger/rdkit_extensions/convert.h"
+#include "schrodinger/rdkit_extensions/file_format.h"
+#include "schrodinger/sketcher/dialog/file_import_export.h"
+#include "schrodinger/sketcher/model/mol_model.h"
 #include "schrodinger/sketcher/model/sketcher_model.h"
 #include "schrodinger/sketcher/ui/ui_paste_in_text_dialog.h"
 
+using ::schrodinger::rdkit_extensions::Format;
+
 namespace schrodinger
 {
 namespace sketcher
 {
+namespace
+{
+
+typedef uint8_t FormatMoleculeType_t;
+namespace FormatMoleculeType
+{
+/**
+ * Whether a given format applies to atomistic models, monomeric models, or both
+ */
+enum : FormatMoleculeType_t {
+    ATOMISTIC = 1 << 0,
+    MONOMERIC = 1 << 1,
+    BOTH = ATOMISTIC | MONOMERIC,
+};
```

Comment:
should we just hijack the existing enum? Seems silly to have them both, though I understand the naming differences https://github.com/schrodinger/sketcher/blob/52c5fd93281429a7a53e2a9521e25e00f0a8c098/include/schrodinger/sketcher/public_constants.h#L32